src/drivers/amd/renoir/ipc.c

```diff
@@ -91,6 +92,9 @@ static void irq_handler(void *arg)
 	if (status) {
 		/* Interrupt source */
 		if (sof_ipc_host_status()) {
+			lock = io_reg_read(PU_REGISTER_BASE + ACP_AXI2DAGB_SEM_0);
+			while (lock)
+				lock = io_reg_read(PU_REGISTER_BASE + ACP_AXI2DAGB_SEM_0);
```

Comment:
This could block FW forever (and make it unrecoverable under some error conditions). Its usually better to have a timeout so if host driver encounters a problem we can unblock here and do a clean error recovery (with responsive FW).

Reply:
Addressed the Review comments.